modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb

```diff
@@ -15,10 +15,11 @@ class PoaFormBuilderJob < ClaimsApi::ServiceBase
       # it queues a job to update the POA code in BGS, as well.
       #
       # @param power_of_attorney_id [String] Unique identifier of the submitted POA
-      def perform(power_of_attorney_id, form_number)
+      def perform(power_of_attorney_id, form_number, rep_id = nil)
         power_of_attorney = ClaimsApi::PowerOfAttorney.find(power_of_attorney_id)
+        rep = ::Veteran::Service::Representative.find(rep_id) unless rep_id.nil?
```

Comment:
Similar thing here, this might not fetch the "right" representative record

Reply:
That should be the rep's primary_key GUID, so it should be the correct record.

Reply:
the primary key for reps is not unique

Reply:
so we should move to something like the controller, correct? using the find_by for all and then checking last updated? using _that_ value?

Reply:
Yeah, ordering a where clause by `created_at: :desc` should ensure it's the same record getting accessed. This should be an edge case. There probably aren't that many rep records with the same rep_id. However, I think for these signatures it matters to get the correct first and last name of the rep.

Reply:
There are 117 rep_ids in staging with more than 1 record